src/test/tests.ts

```diff
@@ -234,6 +241,31 @@ describe('SubscriptionManager', function() {
     });
   });
 
+  it('can use an async filter function', function(done) {
```

Comment:
I don't see it as that. If you revert the changes to `pubsub.ts` this test will fail since and async filter is always truthy even if it ultimately returns a false value. If you have a suggestion about how to restructure the test please let me know.